Automattic/sensei
# Allow Teachers to access Sensei Home

Fixes #5979 
Part of https://github.com/Automattic/sensei-pro/issues/1821
Based on https://github.com/Automattic/sensei/pull/5962

### Changes proposed in this Pull Request

* Opens up the other top level cap (`manage_sensei_grades`) to Sensei Home. 
* On the provider level, restricts access to all providers that aren't News and Guides (for now).
* Expands fetching extensions REST API endpoint with the same permissions. The reason for this is because we aren't lazy loading the `EXTENSIONS_STORE`. In a separate PR (low priority), we should create a hook to only register the store when used.
* Updates tests accordingly.
* This will also benefit our work in [Sensei Blocks](https://github.com/Automattic/sensei-pro/issues/1821) because we can now just bring over the entire `<Main>` component.

### Testing instructions
- As an admin, access Sensei Home and verify that you still can see everything you should.
- As a teacher, access Sensei Home and verify you can see News and Guides.

### Screenshots
![Screen Shot 2022-10-20 at 2 39 20 PM](https://user-images.githubusercontent.com/68693/196964274-90220dd6-7a85-410c-94c4-c239c7cfdc39.png)
